keylime/cmd/create_policy.py

```diff
@@ -151,17 +157,11 @@ def process_ima_buf_in_measurement_list(
                     if path in ignored_keyrings or not get_keyrings:
                         continue
 
-                    if path in keyrings_map:
-                        keyrings_map[path].append(checksum_hash)
-                    else:
-                        keyrings_map[path] = [checksum_hash]
+                    keyrings_map[path] = add_to_list(keyrings_map.get(path), checksum_hash)
                     continue
 
                 if get_ima_buf:
-                    if path in ima_buf_map:
-                        ima_buf_map[path].append(checksum_hash)
-                    else:
-                        ima_buf_map[path] = [checksum_hash]
+                    ima_buf_map[path] = add_to_list(ima_buf_map.get(path), checksum_hash)
```

Comment:
This is only covering part of the places where this can goes wrong. I propose to add a new extra step around line 504, before`jsonpolicy = json.dumps(policy)`, that will normalize the policy removing duplicates. This can be done with something like: `policy["ima_buf"] = { k: sorted(list(set(v))) for k, v in policy["ima-buf"] }`, and do the same for `policy["digest"]`

Reply:
> policy["ima_buf"] = { k: sorted(list(set(v))) for k, v in policy["ima-buf"] }

Sounds good. Add this to the other PR now.

Reply:
@aplanas Thanks.